aws-codeguruprofiler-profilinggroup/src/main/java/software/amazon/codeguruprofiler/profilinggroup/TagHelper.java

```diff
@@ -0,0 +1,148 @@
+package software.amazon.codeguruprofiler.profilinggroup;
+
+import software.amazon.awssdk.services.codeguruprofiler.CodeGuruProfilerClient;
+import software.amazon.awssdk.services.codeguruprofiler.model.CodeGuruProfilerException;
+import software.amazon.awssdk.services.codeguruprofiler.model.ListTagsForResourceRequest;
+import software.amazon.awssdk.services.codeguruprofiler.model.ListTagsForResourceResponse;
+import software.amazon.awssdk.services.codeguruprofiler.model.TagResourceRequest;
+import software.amazon.awssdk.services.codeguruprofiler.model.TagResourceResponse;
+import software.amazon.awssdk.services.codeguruprofiler.model.UntagResourceRequest;
+import software.amazon.awssdk.services.codeguruprofiler.model.UntagResourceResponse;
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
+import software.amazon.cloudformation.proxy.Logger;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static java.util.Collections.emptySet;
+
+public class TagHelper {
+    private static final CodeGuruProfilerClient profilerClient = CodeGuruProfilerClientBuilder.create();
+
+    public static Set<Tag> convertTagMapIntoSet(Map<String, String> tags) {
+        return tags.entrySet()
+                   .stream()
+                   .map(tag ->
+                            Tag.builder().key(tag.getKey()).value(tag.getValue()).build()
+                   )
+                   .collect(Collectors.toSet());
+    }
+
+    public static void updateTags(AmazonWebServicesClientProxy proxy,
+                                  ResourceModel desiredModel,
+                                  String awsAccountId,
+                                  String resourceArn,
+                                  Logger logger) {
+        Set<Tag> existingTags = convertTagMapIntoSet(listTagsForResource(proxy, resourceArn).tags());
+
+        Set<Tag> desiredTags = tagsFromModel(desiredModel);
+
+        if (!existingTags.equals(desiredTags)) {
```

Comment:
How about less indents? Also, add a log when the tags dont need to be updated for easier later debuggin.

```
if (existingTags.equals(desiredTags)) {
    logger.log("No tag change required for... Old: .. New: ... ");
    return;
}

logger.log(...)
```